docs: Add instructions for updating the list of Rook adopters (#12323)
rook/rook: ADOPTERS.md

# Adopters

Below is a list of adopters of Rook in **production environments** that have publicly shared
their usage as well as the benefits provided by Rook that their business relies on.
If you are an adopter of Rook and not yet on this list, we encourage you to add your organization here as well!

> The goal for this list is give inspiration to others that are earlier in their Rook journey.

Reviewer: ```suggestion
The goal for this list is give inspiration to others who have just begun their Rook journey.
```

Reviewer: or 'who are new to Rook`
